cli/crates/common/src/environment.rs

```diff
@@ -109,44 +192,65 @@ impl Environment {
         }
     }
 
-    /// searches for the closest ancestor directory
-    /// named "grafbase" which contains a "schema.graphql" file.
-    /// if already inside a `grafbase` directory, looks for `schema.graphql` inside the current ancestor as well
+    /// searches for the closest ancestor directory named "grafbase" which
+    /// contains either a "grafbase.config.ts" or a "schema.graphql" file. if
+    /// already inside a `grafbase` directory, looks for `schema.graphql` inside
+    /// the current ancestor as well
     ///
     /// # Errors
     ///
     /// returns [`CommonError::ReadCurrentDirectory`] if the current directory path cannot be read
-    fn get_project_grafbase_path() -> Result<Option<PathBuf>, CommonError> {
-        let project_grafbase_path = env::current_dir()
-            .map_err(|_| CommonError::ReadCurrentDirectory)?
-            .ancestors()
-            .find_map(|ancestor| {
-                let mut path = PathBuf::from(ancestor);
-
-                // if we're looking at a directory called `grafbase`, also check for the schema in the current directory
-                if let Some(first) = path.components().next() {
-                    if Path::new(&first) == PathBuf::from(GRAFBASE_DIRECTORY_NAME) {
-                        path.push(GRAFBASE_SCHEMA_FILE_NAME);
-                        if path.is_file() {
-                            return Some(path);
-                        }
-                        path.pop();
+    fn get_project_grafbase_path(warnings: &mut Vec<Warning>) -> Result<Option<GrafbaseSchemaPath>, CommonError> {
```

Comment:
Might make more sense for this function to receive self than mutate an input variable, is it used elsewhere?

Reply:
We don't have an env instance yet to have it as `self`. I'd also have these outside of `Environment` instead of static methods.

Reply:
Hmm, ok, could this maybe return a tuple of result and warnings?
Not sure if it'd be better but I don't love mutating input vars

Reply:
Most of them could be without, but `DateDef` overlaps with `Date`.

Reply:
@pimeys I think you commented on the wrong chain

Reply:
We accumulate warnings, and this is quite often (read: rustc, rust-analyzer) done with a vector. We have more than one warning, so we collect them throughout the run to a collection, then eventually output them.